### PR TITLE
Introducing Unit Tests

### PR DESCRIPTION
**Pull request by lashinijay**

### CODE REVIEW

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
+package org.wso2.carbon.identity.application.authenticator.backupcode;
```
Comment (somindatommy) [inline component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandlerTest.java:19]: Missing license header

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            throws UserStoreException, BackupCodeException {
+
+        mockStatic(BackupCodeUtil.class);
+        mockStatic(MultitenantUtils.class);
+        when(BackupCodeUtil.getUserRealm(username)).thenReturn(userRealm);
+        when(MultitenantUtils.getTenantAwareUsername(username)).thenReturn(tenantAwareUserName);
+        when(BackupCodeUtil.getUserStoreManagerOfUser(username)).thenReturn(userStoreManager);
+        when(userStoreManager.getUserClaimValues(tenantAwareUserName, new String[]{BACKUP_CODES_CLAIM}, null)).
+                thenReturn(userClaimValues);
+        if (username != "test1") {
```
Comment (somindatommy) [inline component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandlerTest.java]: Is there any reason to use `!=` for String comparison?

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandlerTest.java (hunk 1 above)
Comment (somindatommy) [reply component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandlerTest.java:19]: Fix other places as well

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,551 @@
...
+        Map<Integer, StepConfig> stepConfigMap = new HashMap<>();
+        stepConfigMap.put(1,stepConfig);
+
+        SequenceConfig sequenceConfig = new SequenceConfig();
+        sequenceConfig.setStepMap(stepConfigMap);
+
+        AuthenticationContext authenticationContext = new AuthenticationContext();
+        authenticationContext.setSequenceConfig(sequenceConfig);
+
+//        test case #2
```
Comment (somindatommy) [inline component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java]: Is this intentional?

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+        Map<Integer, StepConfig> stepConfigMap2 = new HashMap<>();
+        stepConfigMap2.put(1,stepConfig2);
+
+        SequenceConfig sequenceConfig2 = new SequenceConfig();
+        sequenceConfig2.setStepMap(stepConfigMap2);
+
+        AuthenticationContext authenticationContext2 = new AuthenticationContext();
+        authenticationContext2.setSequenceConfig(sequenceConfig2);
+
+//        test case #3
```
Comment (somindatommy) [inline component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java]: Is this intentional?

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,551 @@
...
+        stepConfig3.setAuthenticatedUser(authenticatedUser3);
+        stepConfig3.setSubjectAttributeStep(true);
+
+        SequenceConfig sequenceConfig3 = new SequenceConfig();
+        sequenceConfig3.setStepMap(null);
+
+        AuthenticationContext authenticationContext3 = new AuthenticationContext();
+        authenticationContext3.setSequenceConfig(sequenceConfig3);
+
+//        test case #4
```
Comment (somindatommy) [inline component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java]: Is this intentional? If so format properly.

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,551 @@
...
+        Map<Integer, StepConfig> stepConfigMap4 = new HashMap<>();
+        stepConfigMap4.put(1,stepConfig4);
+
+        SequenceConfig sequenceConfig4 = new SequenceConfig();
+        sequenceConfig4.setStepMap(stepConfigMap4);
+
+        AuthenticationContext authenticationContext4 = new AuthenticationContext();
+        authenticationContext4.setSequenceConfig(sequenceConfig4);
+
+//        test case #5
```
Comment (somindatommy) [inline component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java]: Is this intentional? If so format properly.

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,551 @@
...
+
+        boolean result = backupCodeUtil.isLocalUser((AuthenticationContext) authenticationContext);
+        assertEquals(expectedVal, result);
+
+    }
+
+    @DataProvider(name = "localUserData")
+    public Object[][] datForLocalUser() {
+
+//        test case #1
```
Comment (somindatommy) [inline component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java:366]: Is this intentional? If so format properly.

Comment (somindatommy) [reply component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java:366]: Fix other places as well

Comment (lashinijay) [reply component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java:366]: addressed with [https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622)

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java (hunk 6 above)
Comment (lashinijay) [reply component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java]: addressed with [https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622)

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java (hunk 5 above)
Comment (lashinijay) [reply component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java]: addressed with [https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622)

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java (hunk 4 above)
Comment (lashinijay) [reply component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java]: addressed with [https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622)

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java (hunk 3 above)
Comment (lashinijay) [reply component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/util/BackupCodeUtilTest.java]: addressed with [https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622)

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandlerTest.java (hunk 2 above)
Comment (lashinijay) [reply component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandlerTest.java]: addressed with [https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622)

File: component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandlerTest.java (hunk 1 above)
Comment (lashinijay) [reply component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandlerTest.java:19]: addressed with [https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/10/commits/aa9856c25ceb3ec01fd73078e83ffe07cf059622)

**Review by somindatommy — APPROVED** [review]

**Pull request merged**